Add resize volume functionality to Logstash operator (#7465)
elastic/cloud-on-k8s: docs/orchestrating-elastic-stack-applications/logstash.asciidoc

[discrete]
== Updating the volume claim settings

> If the storage class allows link:https://kubernetes.io/blog/2018/07/12/resizing-persistent-volumes-using-kubernetes/[volume expansion], you can increase the storage requests size in `spec.volumeClaimTemplates`. ECK will update the existing PersistentVolumeClaims accordingly, and recreate the StatefulSet automatically. If the volume driver supports `ExpandInUsePersistentVolumes`, the filesystem is resized online, without the need of restarting the Logstash process, or re-creating the Pods. If the volume driver does not support `ExpandInUsePersistentVolumes`, Pods must be manually deleted after the resize, to be recreated automatically with the expanded filesystem.

Reviewer: ```suggestion
If the storage class allows link:https://kubernetes.io/blog/2018/07/12/resizing-persistent-volumes-using-kubernetes/[volume expansion], you can increase the storage requests size in `spec.volumeClaimTemplates`. 
ECK updates the existing PersistentVolumeClaims accordingly, and recreates the StatefulSet automatically. 

If the volume driver supports `ExpandInUsePersistentVolumes`, the filesystem is resized online.
In this case, you do not need to restart the {ls} process or re-create the Pods. 

If the volume driver does not support `ExpandInUsePersistentVolumes`, you must manually delete Pods after the resize so that they can be recreated automatically with the expanded filesystem.
```

> Any other changes are forbidden in the volumeClaimTemplates, such as changing the storage class or decreasing the volume size. To make these changes, you have to fully delete the {ls} resource, delete and recreate or resize the volume, and create a new {ls} resource.

Reviewer: ```suggestion
Any other changes in the volumeClaimTemplates--such as changing the storage class or decreasing the volume size--are not allowed. 
To make changes such as these, you must fully delete the {ls} resource, delete and recreate or resize the volume, and create a new {ls} resource.
```

> Before deleting a persistent queue (PQ) volume, ensure that the queue is empty.

Reviewer: ```suggestion
Before you delete a persistent queue (PQ) volume, ensure that the queue is empty.
```
